x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials.test.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import '../../../__mocks__/kea.mock';
+import '../../../__mocks__/shallow_useeffect.mock';
+
+import React, { useEffect } from 'react';
+import { shallow } from 'enzyme';
+import { useValues, useActions } from 'kea';
+
+import { Credentials } from './credentials';
+import { EuiCopy, EuiPageContentBody } from '@elastic/eui';
+
+import { externalUrl } from '../../../shared/enterprise_search_url';
+
+const getUseEffectUnmountHandler = () => (useEffect as jest.Mock).mock.calls[0][0]();
```

Comment:
This is super cool - in fact, it's so cool I'd love to bake it directly into `shallow_useeffect.mock.ts`! Any objection to me doing so in a separate/follow-up PR? 🎉

Reply:
Pulled it out and played around with it here: https://github.com/JasonStoltz/kibana/commit/e2f5a3d922f67b4750866c96abc500969a22cc11

What do you think? 😄

Reply:
It's great Constance, thank you!

Reply:
I will cherrypick this into my branch